samplePDF/Structs.cpp

```diff
@@ -153,18 +150,16 @@ double NoOverflowIntegral(TH2Poly* poly) {
 
 // **************************************************
 //WP: Helper function for projecting TH2Poly onto the X axis
-TH1D* PolyProjectionX(TObject* poly, std::string TempName, std::vector<double> xbins, bool computeErrors) {
+TH1D* PolyProjectionX(TObject* poly, std::string TempName, const std::vector<double>& xbins, const bool computeErrors) {
 // **************************************************
-
-  TH1D* hProjX = new TH1D((TempName+"_x").c_str(),(TempName+"_x").c_str(),xbins.size()-1,&xbins[0]);
+  TH1D* hProjX = new TH1D((TempName+"_x").c_str(),(TempName+"_x").c_str(), xbins.size()-1, &xbins[0]);
 
   //KS: Temp Histogram to store error, use double as this is thread safe
-  double *hProjX_Error = new double[hProjX->GetXaxis()->GetNbins()+1];
-  for (int i = 0; i <= hProjX->GetXaxis()->GetNbins(); ++i) {hProjX_Error[i] = 0;}
-  double xlow, xup, frac=0;
+  std::vector<double> hProjX_Error(hProjX->GetXaxis()->GetNbins() + 1, 0.0);
+  double xlow, xup, frac = 0.;
 
   //loop over bins in the poly
-  for(int i = 0; i<((TH2Poly*)poly)->GetNumberOfBins(); i++)
+  for(int i = 0; i < ((TH2Poly*)poly)->GetNumberOfBins(); i++)
```

Comment:
Can you use a c++ style cast [gonna try and enforce the standard 😆 ]

Reply:
I will try to fix it later but there is flag which give warning if you go with c style casting
https://github.com/mach3-software/MaCh3/blob/963bba0ba4b5596684415cebc0c6b61643ff9221/CMakeLists.txt#L96
this would be best to enforce style